cumulusci/tasks/metadata_etl/picklists.py

```diff
@@ -0,0 +1,185 @@
+from typing import Dict
+
+from cumulusci.core.exceptions import TaskOptionsError
+from cumulusci.tasks.metadata_etl import MetadataSingleEntityTransformTask
+from cumulusci.utils.xml.metadata_tree import MetadataElement
+from cumulusci.core.utils import process_bool_arg, process_list_arg
+
+
+class AddPicklistEntries(MetadataSingleEntityTransformTask):
+    entity = "CustomObject"
+    task_options = {
+        "picklists": {
+            "description": "List of picklists to affect, in Object__c.Field__c form.",
+            "required": True,
+        },
+        "entries": {
+            "description": "Array of picklist values to insert. "
+            "Each value should contain the keys 'fullName', the API name of the entry, "
+            "and 'label', the user-facing label. Optionally, specify `default: True` on exactly one "
+            "entry to make that value the default. Any existing values will not be affected other than "
+            "setting the default (labels of existing entries are not changed).",
+            "required": True,
+        },
+        "record_types": {
+            "description": "List of Record Type developer names for which the new values "
+            "should be available. If any of the entries have `default: True`, they are also made "
+            "default for these Record Types. Any Record Types not present in the target org will be "
+            "ignored, and * is a wildcard. Default behavior is to do nothing."
+        },
+        "sorted": {
+            "description": "Set whether or not to sort the picklist values alphabetically. Default is no change."
+        },
+        **MetadataSingleEntityTransformTask.task_options,
+    }
+
+    def _init_options(self, kwargs):
+        self.task_config.options["api_names"] = "dummy"
+        super()._init_options(kwargs)
+
+        try:
+            if float(self.api_version) < 38.0:
+                raise TaskOptionsError("This task requires API version 38.0 or later.")
+        except ValueError:
+            raise TaskOptionsError(f"Invalid API version {self.api_version}")
+
+        self.picklists = {}
+        for pl in process_list_arg(self.options["picklists"]):
+            try:
+                obj, field = pl.split(".")
+            except ValueError:
+                raise TaskOptionsError(
+                    f"Picklist {pl} is not a valid Object.Field reference"
+                )
+            if not field.endswith("__c"):
+                raise TaskOptionsError(
+                    "This task only supports custom fields. To modify "
+                    "Standard Value Sets, use the add_standard_value_set_entries task."
+                )
+
+            self.picklists.setdefault(self._inject_namespace(obj), []).append(
+                self._inject_namespace(field)
+            )
```

Comment:
You could initialize self.picklists as `defaultdict(lambda: defaultdict(list))` and then this would just be:

    self.picklists[self._inject_namespace(obj)].append(self._inject_namespace(field))